lib/ec2/managers/InstanceManager.rb

```diff
@@ -0,0 +1,459 @@
+require "autoscaling/AutoScaling"
+require "common/manager/Manager"
+require "conf/Configuration"
+require "ec2/EC2"
+require "ec2/loaders/InstanceLoader"
+require "ec2/models/InstanceConfig"
+require "ec2/models/InstanceDiff"
+require "iam/IAM"
+require "util/StatusCodes"
+
+require "aws-sdk"
+require "base64"
+
+module Cumulus
+  module EC2
+    class InstanceManager < Common::Manager
+
+      def initialize
+        super()
+        @create_asset = true
+        @client = Aws::EC2::Client.new(region: Configuration.instance.region, profile: Configuration.instance.profile)
+        @device_name_base = Configuration.instance.ec2.volume_mount_base
+        @device_name_start = Configuration.instance.ec2.volume_mount_start
+        @device_name_end = Configuration.instance.ec2.volume_mount_end
+      end
+
+      def resource_name
+        "EC2 Instance"
+      end
+
+      def local_resources
+        @local_resources ||= Hash[InstanceLoader.instances.map { |local| [local.name, local] }]
+      end
+
+      def aws_resources
+        @aws_resources ||=
+          EC2::named_instances
+            .reject { |name, i| AutoScaling::instance_ids.include?(i.instance_id) }
```

Comment:
Make sure to document this behavior. I assume it's obvious to most people that this is desired, but you might help out that one dumb dude who doesn't think about it.